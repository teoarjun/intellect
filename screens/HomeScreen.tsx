import * as React from 'react';
import {StyleSheet, View, Button} from 'react-native';
import * as Colors from '../assets/styles/Colors';

const HomeScreen = (props: any) => {
  const moveToScreen = () => {
    /* istanbul ignore next */
    props.navigation.navigate('WidgetOne');
  };
  const moveToScreenTwo = () => {
    /* istanbul ignore next */
    props.navigation.navigate('WidgetTwo');
  };
  const button1 = () => {
    return <View style={styles.nextButton}>
    <Button title="Widget 1" color={Colors.WHITE} onPress={moveToScreen} />
  </View>
  }
  const button2 = () =>{
    return <View style={styles.nextButton}>
    <Button title="Widget 2" color={Colors.WHITE} onPress={moveToScreenTwo} />
  </View>
  }
  return (
    <View style={styles.container} testID="calendar">
      <View style={styles.buttonRow}>
        {button1()}
        {button2()}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonRow: {
    flexDirection: 'column',
    marginTop: 10,
  },
  nextButton: {
    marginTop: 20,
    backgroundColor: Colors.BUTTON_COLOR,
    paddingVertical: 5,
    paddingHorizontal: 30,
    borderRadius: 30,
  },
});

export default HomeScreen;
