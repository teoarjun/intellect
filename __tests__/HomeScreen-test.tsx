// Mock out all top level functions, such as get, put, delete and post:
import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../screens/HomeScreen';

it('renders correctly with defaults', () => {
  const navigation = {navigate: jest.fn()};
  const button = renderer
    .create(<HomeScreen name="World" navigation={navigation} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
