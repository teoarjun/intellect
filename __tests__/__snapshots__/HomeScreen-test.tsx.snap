// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="calendar"
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#5EAAB6",
          "borderRadius": 30,
          "marginTop": 20,
          "paddingHorizontal": 30,
          "paddingVertical": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#ffffff",
                },
              ]
            }
          >
            Widget 1
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#5EAAB6",
          "borderRadius": 30,
          "marginTop": 20,
          "paddingHorizontal": 30,
          "paddingVertical": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#ffffff",
                },
              ]
            }
          >
            Widget 2
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
